client/src/state/index.js

```diff
@@ -38,6 +38,17 @@ const requestSlice = createSlice({
     pushRequestList(state, action) {
       state.list = [...state.list, action.payload];
     },
+    changeOneRequest(state, action) {
+      const newRequest = action.payload;
+      const updateRequestList = state.list.map((request) => {
```

Comment:
nit: I prefer ternary here since the logic is quite simple. `state.list.map(request => request === action.payload.id ? old : new)`